pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -600,17 +598,41 @@ func BuildDefaultHTTPRoute(node *model.Proxy, clusterName string, operation stri
 	return defaultRoute
 }
 
+// translatePercentToFractionalPercent translates an v1alpha3 Percent instance
+// to an envoy.type.FractionalPercent instance.
+func translatePercentToFractionalPercent(p *networking.Percent) *xdstype.FractionalPercent {
+	return &xdstype.FractionalPercent{
+		Numerator:   uint32(p.Value * 10000),
+		Denominator: xdstype.FractionalPercent_MILLION,
+	}
+}
+
+// translateIntegerToFractionalPercent translates an int32 instance to an
+// envoy.type.FractionalPercent instance.
+func translateIntegerToFractionalPercent(p int32) *xdstype.FractionalPercent {
+	return &xdstype.FractionalPercent{
+		Numerator:   uint32(p * 10000),
+		Denominator: xdstype.FractionalPercent_MILLION,
+	}
+}
+
 // translateFault translates networking.HTTPFaultInjection into Envoy's HTTPFault
-func translateFault(in *networking.HTTPFaultInjection) *xdshttpfault.HTTPFault {
+func translateFault(node *model.Proxy, in *networking.HTTPFaultInjection) *xdshttpfault.HTTPFault {
 	if in == nil {
 		return nil
 	}
 
 	out := xdshttpfault.HTTPFault{}
 	if in.Delay != nil {
-		out.Delay = &xdsfault.FaultDelay{
-			Type:    xdsfault.FaultDelay_FIXED,
-			Percent: uint32(in.Delay.Percent),
+		out.Delay = &xdsfault.FaultDelay{Type: xdsfault.FaultDelay_FIXED}
+		if util.Is11Proxy(node) {
+			if in.Delay.Percentage != nil {
+				out.Delay.Percentage = translatePercentToFractionalPercent(in.Delay.Percentage)
+			} else {
+				out.Delay.Percentage = translateIntegerToFractionalPercent(in.Delay.Percent)
+			}
+		} else {
+			out.Delay.Percent = uint32(in.Delay.Percent)
```

Comment:
You should add a translateFractionalPercentToInteger and use that value here. Same if else as above (if in.Delay.Percentage...). You simply set the Percent field instead of percentage. 
Istio APIs move independent of Envoy APIs

Reply:
Addressed in 30ce625.